src/hooks/useResizer.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { addActions, actions } from "../actions";
+import { defaultState } from "./useTableState";
+
+defaultState.resizedColumns = {};
+
+addActions({
+  resizeColumn: "__resizeColumn__"
+});
+
+const propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      width: PropTypes.number
+    })
+  ),
+  onResizedChange: PropTypes.func
+};
+
+export const useResizer = props => {
+  PropTypes.checkPropTypes(propTypes, props, "property", "useResizer");
+
+  const {
+    columns,
+    hooks: { headers: headerHooks },
+    state: [{ resizedColumns }, setState],
+    onResizedChange
+  } = props;
+
+  const resizeColumn = (e, intialPosition, initialWidth, columnIndex) => {
+    const { clientX: currentPosition } = e;
+    const positionXDelta = currentPosition - intialPosition;
+    let updatedWidth = initialWidth + positionXDelta > 0 ? initialWidth + positionXDelta : 0;
+
+    if (columns[columnIndex].minWidth && updatedWidth < columns[columnIndex].minWidth) {
+      updatedWidth = columns[columnIndex].minWidth;
+    }
+
+    if (columns[columnIndex].maxWidth && updatedWidth > columns[columnIndex].maxWidth) {
+      updatedWidth = columns[columnIndex].maxWidth;
+    }
+
+    if (updatedWidth !== initialWidth) {
+      setState(old => {
+        let { resizedColumns } = old;
+
+        resizedColumns[columnIndex] = updatedWidth;
+        columns[columnIndex].width = updatedWidth;
+
+        if (onResizedChange) {
+          onResizedChange(resizedColumns, e);
+        }
+
+        return {
+          ...old,
+          resizedColumns
+        };
+      }, actions.resizeColumn);
+    }
+  };
+
+  const getResizer = (columns, api) => {
+    let bottomLevelColumnCount = 0;
+
+    columns.forEach(column => {
+      // if column is not a group (bottom level), add a resizer
+      if (!column.columns) {
+        const index = bottomLevelColumnCount;
+
+        column.resizer = () => (
+          <span
```

Comment:
With the new headless approach, RT hooks should not expose or provide any UI. It would be better to provide a prop getter to the user like `column.getResizerProps(userPropOverrides)` that the user can then use in their own markup.